contrib/generate_instructions_list.jl

```diff
@@ -0,0 +1,40 @@
+using LightXML, JSON
+
+function generate_dict()
+    file = joinpath(@__DIR__, "instructions.xml")
+    # Download the list of instructions from https://uops.info/xml.html
+    if !isfile(file)
+        download("https://uops.info/instructions.xml", file)
+    end
+    xml = parse_file(file)
+    # Get the list of all extensions
+    extensions = get_elements_by_tagname(root(xml), "extension")
+
+    # Accumulator for all instructions.  The strategy is the following: we loop
+    # over the extensions from the lower ones to the higher ones, if an
+    # instruction is already found in a lower extension, then it is not added to
+    # the higher one.
+    all_instructions = String[]
+    dict = Dict{String,Vector{String}}("unknown" => String[], "cpuid" => ["cpuid"])
+    for name in (
+        "MMX", "SSE", "SSE2", "SSE3", "SSSE3", "SSE4", "AVX", "AES", "PCLMULQDQ" , # sandybridge (aka AVX)
+        "MOVBE", "AVX2", "RDWRFSGS", "FMA", "BMI1", "BMI2", "F16C", # haswell (aka AVX2)
+        "PKU", "RDSEED", "ADCX", "CLFLUSH", "XSAVEC", "XSAVES", "CLWB", "AVX512EVEX", "AVX512VEX", # skylake-avx512 (aka AVX512)
+    )
+        instructions = String[]
+        for idx in findall(x -> name == attribute(x, "name"), extensions)
+            for instruction in get_elements_by_tagname(extensions[idx], "instruction")
+                instruction = lowercase(replace(attribute(instruction, "asm"), r"{[a-z]+} " => ""))
+                if instruction ∉ all_instructions
+                    unique!(sort!(push!(all_instructions, instruction)))
+                    unique!(sort!(push!(instructions, instruction)))
+                end
+            end
+        end
+        dict[name] = instructions
+    end
+    free(xml)
+    # We're basically converting an XML to a JSOn, funny isn't it?
+    write(joinpath(@__DIR__, "..", "src", "auditor", "instructions.json"), JSON.json(dict))
```

Comment:
Can we pretty-print the dict?  I know it's more bytes on disk, but it should help a lot, especially with diffs in the future.